OpenRA.Game/ExternalMods.cs

```diff
@@ -107,53 +113,96 @@ internal void Register(Manifest mod)
 				}))
 			};
 
-			var supportPath = Platform.ResolvePath(Path.Combine("^", "ModMetadata"));
+			var locations = new List<string>();
+			if (registration.HasFlag(ModRegistration.System))
+				locations.Add(Platform.SystemSupportDir);
 
-			try
-			{
-				// Make sure the mod is available for this session, even if saving it fails
-				LoadMod(yaml.First().Value);
+			if (registration.HasFlag(ModRegistration.User))
+				locations.Add(Platform.SupportDir);
 
-				Directory.CreateDirectory(supportPath);
-				File.WriteAllLines(Path.Combine(supportPath, key + ".yaml"), yaml.ToLines(false).ToArray());
-			}
-			catch (Exception e)
-			{
-				Log.Write("debug", "Failed to register currrent mod metadata");
-				Log.Write("debug", e.ToString());
-			}
+			// Make sure the mod is available for this session, even if saving it fails
+			LoadMod(yaml.First().Value);
```

Comment:
This is parsing yaml that we have defined just above, and should never be invalid.  It shouldn't be possible to crash here on anything that didn't crash somewhere else earlier.